File: 025-screen-time/screenTime.js
/*
Pseudocode
1. Init answer var as "false"
2. Loop through array
3. Define conditions according to constraints
    IF condition, update answer to true
    IF NOT condition, continue
4. Return answer
*/

function tooMuchScreenTime(arr) {
  const arrSum = arr.reduce((a, b) => a + b);
  let answer = false;

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] >= 10 || i < 5 && ((arr[i] + arr[i + 1] + arr[i + 2]) / 3) >= 8 || (arrSum / 7) >= 6) {
      answer = true;
    }
  }

  return answer;
}

File: 025-screen-time/screen_time.py
'''
Pseudocode
1. Init answer var as "False"
2. Loop through array
3. Define conditions according to constraints
    IF condition, update answer to "True"
    IF NOT condition, continue
4. Return answer
'''

def too_much_screen_time(hour_list):
  answer = False
  for index, element in enumerate(hour_list):
    if element >= 10 or index < 5 and ((hour_list[index] + hour_list[index + 1] + hour_list[index + 2]) / 3) >= 8 or (sum(hour_list) / 7) >= 6:
      answer = True
  return answer

print(too_much_screen_time([1, 2, 3, 4, 5, 6, 7]))
print(too_much_screen_time([7, 8, 8, 4, 2, 2, 3]))
print(too_much_screen_time([5, 6, 6, 6, 6, 6, 6]))
print(too_much_screen_time([1, 2, 3, 11, 1, 3, 4]))
print(too_much_screen_time([1, 2, 3, 10, 2, 1, 0]))
print(too_much_screen_time([3, 3, 5, 8, 8, 9, 4]))
print(too_much_screen_time([3, 9, 4, 8, 5, 7, 6]))